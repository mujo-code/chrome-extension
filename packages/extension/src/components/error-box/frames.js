import { FixedS, FixedL, Box } from '@mujo/ui'
import React from 'react'

const FN_SIZE = 25

const makeNiceFilenames = filename => filename.split('/').pop()
const makeNiceFunctionNames = fnName =>
  `at ${fnName}`.padStart(FN_SIZE, ' ')

export const Frames = ({ frames, error }) => (
  <>
    <FixedL marginTop="xxs" marginBottom="xxs" color="white">
      {error.message}
    </FixedL>
    {frames.map((f, index) => (
      <Box
        marginTop="xxs"
        marginBottom="xxs"
        key={index}
        display="flex"
        flexDirection="row"
      >
        <FixedS
          Component="pre"
          marginTop="zero"
          marginBottom="zero"
          marginRight="m"
          color="saltBox"
        >
          {makeNiceFunctionNames(f.functionName)}
        </FixedS>
        <FixedS marginTop="zero" marginBottom="zero" color="white">
          ({makeNiceFilenames(f.fileName)}:{f.lineNumber})
        </FixedS>
      </Box>
    ))}
  </>
)
