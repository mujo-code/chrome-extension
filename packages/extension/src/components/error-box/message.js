import { BodyS, HeaderS, Box } from '@mujo/ui'
import React from 'react'
import { useTranslation } from 'react-i18next'
import { TRANSLATION_FILE } from '../../constants'

export const Message = ({ errorId }) => {
  const { t } = useTranslation(TRANSLATION_FILE)
  return (
    <>
      <HeaderS maxWidth="500px" color="mischka">
        {t('sorry-for-issue')}
      </HeaderS>
      <BodyS maxWidth="500px" color="mischka">
        {t('created-an-issue')}
        <Box Component="span" color="saltBox">
          ${errorId}
        </Box>
        {t('error-persists')}
      </BodyS>
    </>
  )
}
