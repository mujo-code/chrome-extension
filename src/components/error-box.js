import { css } from '@emotion/core'
import { Box } from '@mujo/box'
import ErrorStackParser from 'error-stack-parser'
import React from 'react'
import { mapStackTrace } from 'sourcemapped-stacktrace'
import { exception } from '../lib/tracker'
import { set } from '../lib/util'
import { FixedS, FixedL } from './fonts'

const FN_SIZE = 25

const makeNiceFilenames = filename => filename.split('/').pop()
const makeNiceFunctionNames = fnName =>
  `at ${fnName}`.padStart(FN_SIZE, ' ')

const ErrorFrames = ({ frames, error }) => (
  <>
    <FixedL marginTop="xxs" marginBottom="xxs" color="white">
      {error.message}
    </FixedL>
    {frames.map((f, index) => (
      <Box
        marginTop="xxs"
        marginBottom="xxs"
        key={index}
        display="flex"
        direction="row"
      >
        <FixedS
          Component="pre"
          marginTop="zero"
          marginBottom="zero"
          marginRight="m"
          color="saltBox"
        >
          {makeNiceFunctionNames(f.functionName)}
        </FixedS>
        <FixedS marginTop="zero" marginBottom="zero" color="white">
          ({makeNiceFilenames(f.fileName)}:{f.lineNumber})
        </FixedS>
      </Box>
    ))}
  </>
)

export class ErrorBox extends React.Component {
  constructor(props) {
    super(props)
    this.state = { hasError: false }
  }

  static getDerivedStateFromError() {
    // Update state so the next render will show the fallback UI.
    return { hasError: true }
  }

  componentDidCatch(error) {
    mapStackTrace(error.stack, stack => {
      set(error, 'stack', stack.join('\n'))
      this.setState({ error })
      exception(error)
    })
  }

  render() {
    const { hasError, error } = this.state
    let frames
    if (error) {
      try {
        frames = ErrorStackParser.parse(error)
      } catch (e) {
        // do nothing
        console.log(e)
      }
    }

    if (hasError) {
      return (
        <Box
          position="fixed"
          backgroundColor="outerSpace"
          padding="xl"
          {...css({
            top: 0,
            left: 0,
            bottom: 0,
            right: 0,
            height: '100vh',
            width: '100vw',
          })}
        >
          <FixedL Component="div" color="saltBox" marginBottom="xl">
            Just like everything, errors are impermanent
          </FixedL>
          {frames && <ErrorFrames frames={frames} error={error} />}
        </Box>
      )
    }

    return <>{this.props.children}</>
  }
}
