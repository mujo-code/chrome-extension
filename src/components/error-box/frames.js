import { Box } from '@mujo/box'
import React from 'react'
import { FixedS, FixedL } from '../fonts'

const FN_SIZE = 25

const makeNiceFilenames = filename => filename.split('/').pop()
const makeNiceFunctionNames = fnName =>
  `at ${fnName}`.padStart(FN_SIZE, ' ')

export const Frames = ({ frames, error }) => (
  <>
    <FixedL marginTop="xxs" marginBottom="xxs" color="white">
      {error.message}
    </FixedL>
    {frames.map((f, index) => (
      <Box
        marginTop="xxs"
        marginBottom="xxs"
        key={index}
        display="flex"
        direction="row"
      >
        <FixedS
          Component="pre"
          marginTop="zero"
          marginBottom="zero"
          marginRight="m"
          color="saltBox"
        >
          {makeNiceFunctionNames(f.functionName)}
        </FixedS>
        <FixedS marginTop="zero" marginBottom="zero" color="white">
          ({makeNiceFilenames(f.fileName)}:{f.lineNumber})
        </FixedS>
      </Box>
    ))}
  </>
)
