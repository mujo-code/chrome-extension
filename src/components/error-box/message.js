import { Box } from '@mujo/box'
import React from 'react'
import { useTranslation } from 'react-i18next'
import { TRANSLATION_FILE } from '../../constants'
import { BodyS, HeaderS } from '../fonts'

export const Message = ({ errorId }) => {
  const { t } = useTranslation(TRANSLATION_FILE)
  return (
    <>
      <HeaderS maxWidth="500px" color="mischka">
        {t('sorry-for-issue')}
      </HeaderS>
      <BodyS maxWidth="500px" color="mischka">
        {t('created-an-issue')}
        <Box Component="span" color="saltBox">
          ${errorId}
        </Box>
        {t('error-persists')}
      </BodyS>
    </>
  )
}
