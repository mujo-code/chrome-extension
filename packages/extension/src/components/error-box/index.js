import { css } from '@emotion/core'
import { HeaderL, Box } from '@mujo/ui'
import ErrorStackParser from 'error-stack-parser'
import React from 'react'
import { ENVIRONMENT } from '../../env'
import { i18n } from '../../i18n'
import { tracker } from '../../lib/error-tracker'
import { Frames } from './frames'
import { Message } from './message'

export class ErrorBox extends React.Component {
  constructor(props) {
    super(props)
    this.state = { hasError: false }
  }

  static getDerivedStateFromError(error) {
    // Update state so the next render will show the fallback UI.
    return { hasError: true, error }
  }

  componentDidCatch(error, errorInfo) {
    tracker.exceptionWithInfo(error, errorInfo).then(errorId => {
      this.setState({ error, errorId })
    })
  }

  render() {
    const { hasError, error, errorId } = this.state
    const isDev = ENVIRONMENT === 'development'
    let frames
    if (error && isDev) {
      try {
        frames = ErrorStackParser.parse(error)
      } catch (e) {
        // do nothing
      }
    }

    if (hasError) {
      return (
        <Box
          position="fixed"
          backgroundColor="outerSpace"
          padding="xl"
          {...css({
            top: 0,
            left: 0,
            bottom: 0,
            right: 0,
            height: '100vh',
            width: '100vw',
          })}
        >
          <HeaderL
            maxWidth="500px"
            Component="div"
            color="saltBox"
            marginBottom="xl"
          >
            {i18n.t('errors-impermanent')}
          </HeaderL>
          {frames && isDev ? (
            <Frames frames={frames} error={error} />
          ) : (
            <Message errorId={errorId} />
          )}
        </Box>
      )
    }

    return <>{this.props.children}</>
  }
}
