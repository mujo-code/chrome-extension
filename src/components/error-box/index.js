import { css } from '@emotion/core'
import { Box } from '@mujo/box'
import ErrorStackParser from 'error-stack-parser'
import React from 'react'
import { useTranslation } from 'react-i18next'
import { TRANSLATION_FILE } from '../../constants'
import { ENVIRONMENT } from '../../env'
import { tracker } from '../../lib/error-tracker'
import { HeaderL } from '../fonts'
import { Frames } from './frames'
import { Message } from './message'

export class ErrorBox extends React.Component {
  constructor(props) {
    super(props)
    this.state = { hasError: false }
  }

  static getDerivedStateFromError(error) {
    // Update state so the next render will show the fallback UI.
    return { hasError: true, error }
  }

  componentDidCatch(error, errorInfo) {
    tracker.exceptionWithInfo(error, errorInfo).then(errorId => {
      this.setState({ error, errorId })
    })
  }

  render() {
    const { hasError, error, errorId } = this.state
    const { t } = useTranslation(TRANSLATION_FILE)
    const isDev = ENVIRONMENT === 'development'
    let frames
    if (error && isDev) {
      try {
        frames = ErrorStackParser.parse(error)
      } catch (e) {
        // do nothing
      }
    }

    if (hasError) {
      return (
        <Box
          position="fixed"
          backgroundColor="outerSpace"
          padding="xl"
          {...css({
            top: 0,
            left: 0,
            bottom: 0,
            right: 0,
            height: '100vh',
            width: '100vw',
          })}
        >
          <HeaderL
            maxWidth="500px"
            Component="div"
            color="saltBox"
            marginBottom="xl"
          >
            {t('errors-impermanent')}
          </HeaderL>
          {frames && isDev ? (
            <Frames frames={frames} error={error} />
          ) : (
            <Message errorId={errorId} />
          )}
        </Box>
      )
    }

    return <>{this.props.children}</>
  }
}
