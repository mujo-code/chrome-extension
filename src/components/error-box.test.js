import React from 'react'
import { create, act } from 'react-test-renderer'
import { ErrorBox } from './error-box'

jest.mock('sourcemapped-stacktrace')

test('ErrorBox should match snapshot', () => {
  const tree = create(<ErrorBox />).toJSON()
  expect(tree).toMatchSnapshot()
})

test('ErrorBox should match when there is an error', () => {
  const ErrorComponent = () => {
    throw new Error('foo')
  }
  const wrapper = create(
    <ErrorBox>
      <ErrorComponent />
    </ErrorBox>
  )
  act(() => {
    wrapper.update()
  })
  expect(wrapper.toJSON()).toMatchSnapshot()
})
