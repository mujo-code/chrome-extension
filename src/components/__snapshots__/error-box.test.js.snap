// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBox should match snapshot 1`] = `null`;

exports[`ErrorBox should match when there is an error 1`] = `null`;
